### ISSUE TITLE
Spark of Life not proccing on heals

### ISSUE THREAD

**Issue by Azu3652** [Duskhaven/Bug-Reports#450]
Spark of Life trinket from Keystone Master vendor is not proccing.

**Comment by harukaze3** [comment]
Expanding on this: The Trinket has an On-Hit instead of an On-Equip effect, this is most likely responsible for it not proccing. Although I've been spamming an AOE ability for 2 minutes, and never once saw it, so there might be something else as well.

> GM Haruka

**Comment by Dova14** [comment]
Upon further testing: At training dummy area and doing rotation as enhance it started to proc, afterwards I tried seeing if its something in resto wasnt proccing it.  I spent all 51 points all the way to riptide and it procc'd.  Upon leaving the training dummy area it stopped proccing.  what this means? I do not know

**Comment by Dova14** [comment]
I believe the "On-Hit" affect Haruka mentions is putting up an invisible flag for the trinket to start proccing

**Comment by Dova14** [comment]
So I think I figured it out, Its 100% the "On-hit" instead of equip that is causing an invisible flag to make it start proccing as enhance (Aka the Auto Attacks).  After my last test I went to a dummy and just pressed lesser healing wave on myself and flame shocked on cd:  Had 87% uptime or 10 procs within 1:33.  After unequipping and re-equipping: It had 0% uptime within 1:37 doing the same thing as last time.  
https://imgur.com/a/Wl89kNj

The invisible Flag lasts through zoning, but does not last through unequipping and re-equipping it.  So bandaid fix is to auto attack dummies until you get the flag, will test if it lasts through death
Update: Flag lasts through death
